lib/rucio/client/commands/bin_legacy/rucio.py

```diff
@@ -1631,30 +1631,27 @@ def list_rules_history(args, client, logger, console, spinner):
     for rule in client.list_replication_rule_full_history(scope, name):
         if rule['rule_id'] not in rule_dict:
             rule_dict.append(rule['rule_id'])
-            if cli_config == 'rich':
-                table_data.append(['Insertion', rule['account'], rule['rse_expression'], rule['created_at']])
-            else:
-                print('-' * 40)
-                print('Rule insertion')
-                print('Account : %s' % rule['account'])
-                print('RSE expression : %s' % (rule['rse_expression']))
-                print('Time : %s' % (rule['created_at']))
+            table_data.append(['Insertion', rule['account'], rule['rse_expression'], rule['created_at']])
         else:
             rule_dict.remove(rule['rule_id'])
-            if cli_config == 'rich':
-                table_data.append(['Deletion', rule['account'], rule['rse_expression'], rule['updated_at']])
-            else:
-                print('-' * 40)
-                print('Rule deletion')
-                print('Account : %s' % rule['account'])
-                print('RSE expression : %s' % (rule['rse_expression']))
-                print('Time : %s' % (rule['updated_at']))
+            table_data.append(['Deletion', rule['account'], rule['rse_expression'], rule['updated_at']])
 
-    if cli_config == 'rich':
+    if args.csv:
+        print('ACTION', 'ACCOUNT', 'RSE EXPRESSION', 'TIME', sep=',')
```

Comment:
‘CSV’ stands for ‘comma-separated values’, but it’s an ill-defined format, with a tab or a semi-colon being frequent choices for a separator. Choosing the comma will end up being problematic with `list-identities`.

In addition, I think it wise that the separator be passed not as an individual string per `print()` call, but as a global variable (e.g. `CSV_SEPARATOR`).